app/modules/site_settings/helpers.py

```diff
@@ -37,7 +37,12 @@ def validate_fields(dictionary, fields, error_str):
             if not isinstance(dictionary[field], field_type):
                 raise HoustonException(log, f'{field} incorrect type in {error_str}')
 
-        if field_type == str and field in dictionary and len(dictionary[field]) == 0:
+        if (
```

Comment:
Ah that's where the bug was, apologies for that and good catch. 
The question is, is this really needed. 
The FE sends empty strings for some (non custom field config) fields so we need to deal with them and that validation step got removed. I'm not sure but I think it may actually be a correct check in this situation. Thoughts?

Reply:
yeah, the "what can be set as empty-string" thing is _nearly_ as much a problem as "what can be a null".  :woozy_face: 
i will have to think a little more about this.  or wait for some counterexample to show up that breaks things, _then_ address?

Reply:
yeah something does seem off here.  i suspect this might need slack conversation.  but i am staring at 2 cases (mandatory and not) and trying to walk thru.  like from species:
1. `('scientificName', str, True)`  (mandatory, str)
2. `('itisTsn', int, False)` (not mandatory but also not str)
3. `('fake', str, False)`  (not mandatory, str -- and i made it up)

this last bit of code in question is (i think!) meant to cache the case where the _type_ of the dict value is (correctly) `str` but the string is empty.  if the field is _not_ mandatory (say, like `description` on customFields or `fake` in my example) do we care if it gets set to an empty string?  i dont know.  it seems pretty dumb to me (description would show up as `''`).  but if we want to block that specific case, yes i think this needs retooling.

would this be a fix?
```
if field_type == str and field in dictionary and dictionary[field] is not None and dictionary[field] == '':
```
i think that let thru the unset value (for non-mandatory things like description if they happen to be set `None`) but blocks _anything_ set to empty-string (mandatory or not).  _if_ that is what we want.   omg this is convoluted.  haha

Reply:
I think the best thing to do is leave as is for now. The FE sends some fields as '' if the user does not populate them so any further changes should be post this PR.